billy-spain/src/main/java/com/premiumminds/billy/spain/persistence/dao/DAOESCreditNoteEntry.java

```diff
@@ -18,12 +18,12 @@
  */
 package com.premiumminds.billy.spain.persistence.dao;
 
+import com.premiumminds.billy.core.persistence.entities.GenericInvoiceEntity;
 import com.premiumminds.billy.spain.persistence.entities.ESCreditNoteEntity;
 import com.premiumminds.billy.spain.persistence.entities.ESCreditNoteEntryEntity;
-import com.premiumminds.billy.spain.services.entities.ESInvoice;
 
 public interface DAOESCreditNoteEntry extends AbstractDAOESGenericInvoiceEntry<ESCreditNoteEntryEntity> {
 
-    public ESCreditNoteEntity checkCreditNote(ESInvoice invoice);
+    public ESCreditNoteEntity checkCreditNote(GenericInvoiceEntity invoice);
```

Comment:
Why should we want to allow the possibility of the method to receive any type? Is it due to simple_invoices credit also being a credit_note?